services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/TableManagementStore.java

```diff
@@ -651,94 +726,94 @@ public Future<Versioned<Boolean>> setCredentials(
-                            )
-
-                            // then create new entries
-                            .compose(updatedCredentials -> {
-                                updatedCredentials.createMissingSecretIds();
-                                return CompositeFuture.all(updatedCredentials.getData().stream()
-                                    .map(JsonObject::mapFrom)
-                                    .filter(c -> c.containsKey("type") && c.containsKey("auth-id"))
-                                    .map(c -> this.insertCredentialEntryStatement
-                                            .expand(map -> {
-                                                map.put("tenant_id", key.getTenantId());
-                                                map.put("device_id", key.getDeviceId());
-                                                map.put("type", c.getString("type"));
-                                                map.put("auth_id", c.getString("auth-id"));
-                                                map.put("data", c.toString());
-                                            })
-                                            .trace(this.tracer, span.context())
-                                            .update(connection))
-                                    .collect(Collectors.toList()))
-                                    .mapEmpty();
-                            })
-
-                            // update the version, this will release the lock
-                            .compose(x -> this.updateDeviceVersionStatement
-                                    .expand(map -> {
-                                        map.put("tenant_id", key.getTenantId());
-                                        map.put("device_id", key.getDeviceId());
-                                        map.put("expected_version", version);
-                                        map.put("next_version", nextVersion);
-                                    })
-                                    .trace(this.tracer, span.context())
-                                    .update(connection)
+                        readDeviceForUpdate(connection, key, context)
 
-                                    // check the update outcome
-                                    .compose(TableManagementStore::checkUpdateOutcome))
+                                // check if we got back a result, if not this will abort early
+                                .compose(result -> extractVersionForUpdate(result, resourceVersion))
+
+                                // take the version and start processing on
+                                .compose(version -> Future.succeededFuture()
+
+                                        .compose(x -> {
+                                            final Promise<CredentialsDto> result = Promise.promise();
+                                            final var updatedCredentialsDto = CredentialsDto.forUpdate(
```

Comment:
is anything being changed here or did you only change the indentation?

Reply:
only the indentation